mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h

```diff
@@ -341,8 +345,18 @@ class PopulateParamsBase {
     if (!(dataPerThreadCopy > 0))
       return mlir::failure();
 
+    // FIXME: vectorizationSize is fixed to begin from 4 for now.
+    // Change ConvolutionContext so it carries data type information
+    // so we can possibly use 8 in case of f16.
+    int64_t vectorizationSize = 4;
+    // FIXME: set vectorizationSize be 1 for backward data for now.
+    // The logic for deciding vectorization size and dimension for
+    // backward data has to be reviewed.
+    auto opType = ctx.opType;
+    if (opType == mlir::miopen::ConvOpType::Conv2DBwdDataOpType) {
+      vectorizationSize = 1;
```

Comment:
vectorizationSize is 1, so it should be safe.

This function (PopulateParamsBase) doesn't have much to be reviewed for now.

obtainGemmADimKVectorizable, etc and vectorizationSize need to be updated in future PR

Reply:
we will need to review those logic and file subsequent PRs.